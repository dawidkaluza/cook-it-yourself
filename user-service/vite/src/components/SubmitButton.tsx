import React from "react";

export const SubmitButton = (props: {
  loading?: boolean,
  style?: string
  children: React.ReactNode;
}) => {
  return (
    <button
      type="submit"
      disabled={props.loading}
      className={
        "bg-blue-500 hover:bg-blue-600 text-white font-bold py-2 px-6 rounded disabled:opacity-75 disabled:hover:bg-blue-500 "
        + (props.style ?? "")
      }
    >
      {props.children}
    </button>
  );
};