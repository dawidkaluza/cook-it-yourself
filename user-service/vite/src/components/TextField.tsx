import {ChangeEventHandler, useId} from "react";

export const TextField = (props: {
  name: string;
  label: string;
  type?: "text" | "password";
  value?: string;
  onChange?: ChangeEventHandler<HTMLInputElement>;
  placeholder?: string;
  style?: string;
}) => {
  const id = useId();
  return (
    <div className={props.style}>
      <label htmlFor={props.name + "_" + id} className="block mb-1 font-medium">
        {props.label}
      </label>
      <input
        id={props.name + "_" + id}
        name={props.name}
        type={props.type}
        value={props.value}
        onChange={props.onChange}
        placeholder={props.placeholder}
        className="w-full py-2 px-3 rounded border shadow text-gray-700 outline-none focus:border-blue-400"
      />
    </div>
  );
};