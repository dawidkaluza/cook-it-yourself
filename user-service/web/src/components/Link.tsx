import React from "react";
import {Link as RouterLink} from "react-router-dom";

export const Link = (props: {
  to: string;
  children: React.ReactNode;
}) => {
  return (
    <RouterLink
      to={props.to}
      className="font-bold text-blue-500 hover:text-blue-700"
    >
      {props.children}
    </RouterLink>
  );
};