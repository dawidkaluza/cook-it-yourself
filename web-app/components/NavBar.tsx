import React from "react";
import Link from "next/link";

const NavBar = () => {
  return (
    <nav className="navbar navbar-expand bg-body-tertiary">
      <div className="container-fluid text-center">
        <Link href="/" className="navbar-brand">Cook it yourself</Link>
        <Link href="/sign-in" className="btn btn-primary btn-sm">Sign in</Link>
      </div>
    </nav>
  );
};

export { NavBar };