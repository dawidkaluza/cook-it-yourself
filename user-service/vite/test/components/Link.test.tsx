import {render, screen} from "@testing-library/react";
import {expect} from "vitest";
import {Link} from "../../src/components/Link";
import {MemoryRouter} from "react-router-dom";

describe("Link component", () => {
  test("render", () => {
    // Given
    // When
    const component = render(
      <MemoryRouter>
        <Link to={"/about-me"}>About me</Link>
      </MemoryRouter>
    );

    // Then
    const aEl = screen.getByRole("link", { name: "About me" });
    expect(aEl).toBeDefined();
    expect(aEl.getAttribute("href")).toBe("/about-me");

    expect(component.container).toMatchSnapshot();
  });
})