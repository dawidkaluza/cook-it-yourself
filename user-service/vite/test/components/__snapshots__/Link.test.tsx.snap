// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Link component > render 1`] = `
<div>
  <a
    class="font-bold text-blue-500 hover:text-blue-700"
    href="/about-me"
  >
    About me
  </a>
</div>
`;
